src/Dialect/ONNX/ONNXOps/NN/RotaryEmbedding.cpp

```diff
@@ -0,0 +1,117 @@
+// (c) Copyright 2022 - 2025 Advanced Micro Devices, Inc. All Rights Reserved.
+
+// =============================================================================
+//
+// This file provides definition of ONNX dialect RotaryEmbedding operation.
+//
+//===----------------------------------------------------------------------===//
+
+#include "ONNXOps.hpp"
+
+#include "src/Dialect/ONNX/ONNXOps/OpHelper.hpp"
+
+using namespace mlir;
+using namespace mlir::OpTrait::util;
+using namespace onnx_mlir;
+
+namespace onnx_mlir {
+
+template <>
+LogicalResult ONNXRotaryEmbeddingOpShapeHelper::computeShape() {
+  ONNXRotaryEmbeddingOpAdaptor operandAdaptor(operands);
+  return setOutputDimsFromOperand(operandAdaptor.getX());
+}
+
+} // namespace onnx_mlir
+
+//===----------------------------------------------------------------------===//
+// Verify
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXRotaryEmbeddingOp::verify() {
+  ONNXRotaryEmbeddingOpAdaptor adaptor(*this);
+  Value input = adaptor.getX();
+  if (!hasShapeAndRank(input))
+    return success(); // Won't be able to do any checking at this stage.
+
+  auto inputType = mlir::cast<ShapedType>(input.getType());
+  int64_t inputRank = inputType.getRank();
```

Comment:
Tiny nit, feel free to ignore
```suggestion
  const int64_t inputRank = inputType.getRank();
```